src/routes/clients.js

```diff
@@ -31,13 +33,26 @@ let router = express.Router();
 // Create a new API client.
 router.post('/', (req, res) => {
   // Required params: name
-  // XXX We will probably require redirect url for signin/signup flow.
   req.checkBody('name', 'missing or invalid required "name" parameter')
      .notEmpty();
 
-  const errors = req.validationErrors();
-  if (errors) {
-    return ApiError(res, 400, ERRNO_INVALID_API_CLIENT_NAME, BAD_REQUEST);
+  if (req.body.redirectUrl) {
+    req.checkBody('redirectUrl', 'invalid "redirectUrl"')
+       .isUrl({ require_valid_protocol: true });
+  }
+
+  const error = req.validationErrors()[0];
+  if (error) {
+    let errno = ERRNO_BAD_REQUEST;
+    switch (error.param) {
+      case 'redirectUrl':
+        errno = ERRNO_INVALID_API_CLIENT_REDIRECT_URL;
+        break;
+      case 'name':
+        errno = ERRNO_INVALID_API_CLIENT_NAME;
+        break;
```

Comment:
nit: I'd find it more readable to use a `default:` close to set `ERRNO_BAD_REQUEST`. (+ eslint will be happier when I'll set it up :) )